src/actions/callsuborchestratoraction.ts

```diff
@@ -0,0 +1,20 @@
+import { ActionType, IAction } from "../classes";
+
+/** @hidden */
+export class CallSubOrchestratorAction implements IAction {
+    public readonly actionType: ActionType = ActionType.CallSubOrchestrator;
+
+    constructor(
+        public readonly functionName: string,
+        public readonly instanceId?: string,
+        public readonly input?: unknown,
+    ) {
+        if (!functionName || typeof functionName !== "string") {
+            throw new TypeError(`functionName: Expected non-empty string but got ${typeof functionName}`);
+        }
+
+        if (instanceId && typeof instanceId !== "string") {
+            throw new TypeError(`isntanceId: Expected non-empty string but got ${typeof instanceId}`);
+        }
```

Comment:
It seems there are a lot of redundant checks. Could these be factored out into common functions?

Reply:
This could definitely be done as an optimization; I can put them in Utils.